src/MentionsInput.js

```diff
@@ -541,15 +541,42 @@ const MentionsInput = React.createClass({
     // neglect async results from previous queries
     if(queryId !== this._queryId) return;
 
+    var type = mentionDescriptor.props.type;
     var update = {};
-    update[mentionDescriptor.props.type] = {
-      query: query,
-      mentionDescriptor: mentionDescriptor,
-      querySequenceStart: querySequenceStart,
-      querySequenceEnd: querySequenceEnd,
-      results: suggestions,
-      plainTextValue: plainTextValue
-    };
+
+    // If mention doesn't specify type try suggestion type
+    if(!type) {
+      var groups = {}
```

Comment:
Why is this necessary? 

In my mind, the purpose of splitting the `suggestions` state by mentions type is only to track whether results for different async suggestion queries are still valid (i.e., the query did not change in the meanwhile). Since in your scenario there's only one `Mention` source, it should not be necessary to track different states per group.

I'd expect, you could simply leave this method unchanged, no?